ocaml/xapi/xapi_vgpu_type.ml

```diff
@@ -541,173 +469,202 @@ module Intel = struct
         else [passthrough_gpu]
       in
       passthrough_types @
-      (make_vgpu_types ~__context ~pci ~whitelist:!Xapi_globs.gvt_g_whitelist)
+      (make_vgpu_types ~__context ~pci ~whitelist:V.whitelist_file)
     in
     List.map (find_or_create ~__context) types
-end (* Intel *)
+end
 
-module MxGPU = struct
-  let amd_vendor_id = 0x1002
+module Vendor_intel = struct
+  type vgpu_conf = {
+    identifier : Identifier.gvt_g_id;
+    experimental : bool;
+    model_name : string;
+    framebufferlength : int64;  (* in MiB *)
+    num_heads : int64;
+    max_x : int64;
+    max_y : int64;
+  }
 
-  let ( *** ) = Int64.mul
-  let ( /// ) = Int64.div
-  let ( +++ ) = Int64.add
-  let ( --- ) = Int64.sub
-  let mib x = List.fold_left Int64.mul x [1024L; 1024L]
+  let vendor_id = 0x8086
+  let whitelist_file = !Xapi_globs.gvt_g_whitelist
+  let device_id_of_conf conf = conf.identifier.Identifier.pdev_id
 
+  let read_whitelist_line ~line =
+    try
+      Some (Scanf.sscanf
+              line
+              "%04x experimental=%c name='%s@' low_gm_sz=%Ld high_gm_sz=%Ld fence_sz=%Ld framebuffer_sz=%Ld max_heads=%Ld resolution=%Ldx%Ld monitor_config_file=%s"
```

Comment:
The whole MxGPU module still fits on one screen though.
But maybe not on your screen with your font. Ah well, I don't mind.